ethereumj-core/src/main/java/org/ethereum/sharding/proposer/ProposerServiceImpl.java

```diff
@@ -68,17 +75,25 @@ public ProposerServiceImpl(BeaconProposer proposer, BeaconChain beaconChain,
     }
 
     @Override
-    public void init(BeaconState state) {
+    public void init(BeaconState state, byte[]... pubKeys) {
+        assert pubKeys.length > 0;
+
+        this.validatorIndices = new HashSet<>();
+
+        for (byte[] pubKey : pubKeys) {
+            Validator validator = state.getValidatorSet().getByPubKey(pubKey);
+            if (validator != null) {
+                this.validatorIndices.add(validator.getIndex());
+                this.pubKeysMap.put(validator.getIndex(), validator.getPubKey());
+            } else {
+                // something went wrong
+                logger.error("Failed to start proposer for {}: validator does not exist", HashUtil.shortHash(pubKey));
+                return;
+            }
 
-        Validator validator = state.getValidatorSet().getByPupKey(config.pubKey());
-        // something went wrong
-        if (validator == null) {
-            logger.error("Failed to start proposer: validator {} does not exist", Hex.toHexString(config.pubKey()));
-            return;
+            this.validatorIndices.add(validator.getIndex());
```

Comment:
why this one is needed?

Reply:
I mean
```
(86th line)                 this.validatorIndices.add(validator.getIndex());
(94th line)                 this.validatorIndices.add(validator.getIndex());
```
Looks strange in diff

Reply:
Good catch 👍

Reply:
@zilm13 What do you think about this update https://github.com/ethereum/ethereumj/commit/a2030c4261e13d1eae49ec8f5758def0cb9fc65b?

Reply:
looks like redundancy removed, cool!